src/ghga_event_schemas/validation.py

```diff
@@ -47,7 +55,22 @@ def get_validated_payload(payload: JsonObject, schema: type[Schema]) -> Schema:
     try:
         return schema(**payload)
     except pydantic.ValidationError as error:
-        raise EventSchemaValidationError(payload=payload, schema=schema) from error
+        errors = error.errors(
+            include_context=False, include_url=False, include_input=False
+        )
+        missing = [str(e["loc"][0]) for e in errors if e["type"] == "missing"]
+        mistyped = {
+            str(e["loc"][0]): e["msg"] for e in errors if e["type"] != "missing"
+        }
+        unexpected = [_ for _ in payload if _ not in schema.model_fields]
+        error_info = SchemaErrorInfo(
+            missing_fields=missing,
+            mistyped_fields=mistyped,
+            unexpected_fields=unexpected,
+        )
```

Comment:
Are you sure this catches all possible error types?

If you just include the `str(error)` in the message, wouldn't that be sufficient?

Reply:
I was going for something more concise, but have changed it.

Reply:
My expectation was that the error message is concise enough but I can be wrong.

Reply:
It is regardless better than what was displayed previously.

Reply:
We should check this by adding test cases at least for the three error types you had covered before. I.e. use some pattern matching to verify that the expected info is part of the error message.

Reply:
As far as I see this would not happen anyway since we haven't set extra=forbid, right?

Reply:
I forgot about the existence of that. In the event that we do, it will show up in the error message in the same format as other field validation messages, with `Extra inputs are not permitted [type=extra_forbidden...]`. Because we're directly reprinting the error message from pydantic, any issues should come from pydantic's side.